engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1203,17 +1209,33 @@ private AllocateResponse sendContainerAskToRM(List<ContainerRequest> containerRe
       amRmClient.releaseAssignedContainer(containerId);
     }
 
+    return amRmClient.allocate(0);
+  }
+
+  private void processContainerStopRequests()
+  {
     for (String containerIdStr : dnmgr.containerStopRequests.values()) {
       AllocatedContainer allocatedContainer = this.allocatedContainers.get(containerIdStr);
       if (allocatedContainer != null && !allocatedContainer.stopRequested) {
         nmClient.stopContainerAsync(allocatedContainer.container.getId(), allocatedContainer.container.getNodeId());
         LOG.info("Requested stop container {}", containerIdStr);
         allocatedContainer.stopRequested = true;
       }
+
+      Long pendingSince = pendingContainersToKill.get(containerIdStr);
```

Comment:
Is a separate logic block needed for this? Can't the addition to the pendingContainersToKill go into the previous if block.